test/unit/path_spec.lua

```diff
@@ -366,134 +366,150 @@ describe('path.c', function()
   end)
 
   describe('vim_FullName', function()
-    local function vim_FullName(filename, buf, len, force)
-      filename = to_cstr(filename)
-      return cimp.vim_FullName(filename, buf, len, force)
+    -- These variables get reset in before_each().
+    local buf
+    local do_expand
+
+    local function vim_FullName(path, buflen)
+      buf = cstr(buflen, '')
+      return cimp.vim_FullName(to_cstr(path), buf, buflen, do_expand)
+    end
+
+    local function get_buf_len(s, t)
+      return math.max(string.len(s), string.len(t)) + 1
     end
 
     before_each(function()
       -- Create empty string buffer which will contain the resulting path.
-      length = (string.len(lfs.currentdir())) + 33
-      buffer = cstr(length, '')
+      buf = cstr(0, '')
+      do_expand = 1
     end)
 
     itp('fails if given filename is NULL', function()
-      local force_expansion = 1
-      local result = cimp.vim_FullName(NULL, buffer, length, force_expansion)
+      local result = cimp.vim_FullName(NULL, buf, 10, do_expand)
       eq(FAIL, result)
     end)
 
     itp('fails safely if given length is wrong #5737', function()
-      local force_expansion = 1
-      local filename = 'foo/bar/bazzzzzzz/buz/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/a'
+      local path = 'foo/bar/bazzzzzzz/buz/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/a'
```

Comment:
I intentionally avoid naming variables "path" (usually) since (1) it often means $PATH and (2) "filename" tends to be more common in most APIs I've seen. Sometimes `filepath` is ok in contexts where "filename" means only the tail.